### DISCUSSION TITLE
Blockchain Project In Scratch

### DISCUSSION THREAD

**Discussion by HappyPr0gramming** [community/community#129743] [Programming Help]
### Select Topic Area

Question

### Body

Hello, I am a programmer who knows Python, HTML, and is learning Lua. I was playing around in Scratch (scratch.mit.edu) and I wanted to see if making a whole cryptocurrency is possible in Scratch. I have already made a blockchain mining system but I am struggling on how to make a wallet and transaction system. Anyone who can help?

<img width="360" alt="Screenshot 2024-06-25 at 19 11 58" src="https://github.com/community/community/assets/156795708/eba716bd-59e9-4ac3-9592-601efa075363">

**Comment by whisperky** [comment]
Great job on the mining system! Here’s a simple approach to add wallets and transactions:

## 1. Wallets
Use two lists: Wallets to store unique wallet IDs, and Balances to store each wallet’s balance. Each wallet ID's index in Wallets matches its balance in Balances.
## 2. Transactions
Inputs: Ask for the sender’s ID, receiver’s ID, and amount.
Validation: Check if the sender has enough balance.
Processing: Subtract the amount from the sender's balance and add it to the receiver's balance if valid.
### Example code
```
if <(item (senderID) of [Balances]) >= (amount)> then
    change item (senderID) of [Balances] by (0 - amount)
    change item (receiverID) of [Balances] by (amount)
    say [Transaction Successful] for (2) seconds
else
    say [Insufficient Balance] for (2) seconds
end
```
## 3. Blockchain Update
Store each transaction in a Transactions list to simulate adding to the blockchain.

This structure keeps it simple but functional. 
I hope this will help you. Feel free to let me know if you need more details!
